src/Features/Core/Portable/Common/TaggedText.cs

```diff
@@ -88,24 +85,19 @@ public override string ToString()
 
     internal static class TaggedTextExtensions
     {
-        public static ImmutableArray<TaggedText> ToTaggedText(this IEnumerable<SymbolDisplayPart> displayParts, Func<ISymbol, string> getNavigationHint = null, bool includeNavigationHints = true)
-            => displayParts.ToTaggedText(TaggedTextStyle.None, getNavigationHint, includeNavigationHints);
-
         public static ImmutableArray<TaggedText> ToTaggedText(
-            this IEnumerable<SymbolDisplayPart> displayParts, TaggedTextStyle style, Func<ISymbol, string> getNavigationHint = null, bool includeNavigationHints = true)
+            this IEnumerable<SymbolDisplayPart>? displayParts, TaggedTextStyle style = TaggedTextStyle.None, bool includeNavigationHints = true)
         {
             if (displayParts == null)
                 return [];
 
-            getNavigationHint ??= static symbol => symbol?.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
-
             return displayParts.SelectAsArray(d =>
                 new TaggedText(
                     GetTag(d),
                     d.ToString(),
                     style,
                     includeNavigationHints && d.Kind != SymbolDisplayPartKind.NamespaceName ? GetNavigationTarget(d.Symbol) : null,
-                    includeNavigationHints && d.Kind != SymbolDisplayPartKind.NamespaceName ? getNavigationHint(d.Symbol) : null));
+                    includeNavigationHints && d.Kind != SymbolDisplayPartKind.NamespaceName ? d.Symbol?.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat) : null));
```

Comment:
the intent here was to defer this work until needed.  it's a bit concerning that this would now all be done up front.

Reply:
I don't understand. How was the work being deferred before? The lambda is directly invoked here. I just inlined it. Seems like the intent was actually to customize the formatting of the hint, but the only place that did it ended up calling the same ToDisplayString.

Reply:
touche.  i misunderstood that :)

Reply:
> and it would need to implement ToDisplayString. Would that make sense though?

What's a case where we end up calling ToDisplayString on a CodeGenSYmbol?  

> Currently the CodeGenerationSymbol isn't tied to a language and it would need to implement ToDisplayString. Would that make sense though?


So, the main issue is that display strings are literally lang specific.  THat's why the impls in C# and VB then call into the static public helpers (like `SymbolDisplay.ToDisplayString(sym)`).  So in the cases where we need the display string and we potentially take an arbitrary symbol that could be a codegen symbol, we call into those helpers directly (since they support symbols coming in from the other lang).

We could have a way for the caller to pass in which lang to call through using a lang service of some sort though.